### ISSUE TITLE
Cloudant replication too long 

### ISSUE THREAD

**Issue by veetil09** [pouchdb/pouchdb#6917]
### Issue
Replicating a remote cloudant DB to a local in-memory pouch DB takes too much time irrespective of the number of documents fetched.

### Info
- Environment: Node.js
- Platform: Docker
- Adapter: pouchdb-memory
- Server: Cloudant

### Reproduce
```Javascript
const PouchDbMemory = require('pouchdb-memory');
const inmemeorydb = new PouchDbMemory(<remote-cloudant-db-name>);
inmemeorydb.replicate.from(
                <remote-cloudant-db-url>,
                {
                   filter: <filter-in-cloudant-design-doc>
            }).on('change', function (info) {
                logger.info(' UPDATED ', info.docs.length);
            }).on('active', function () {
                logger.info('ACTIVE');
            }).on('complete', function (info) {
                logger.info('COMPLETE: ', JSON.stringify(info, null, 2));
           });
```

Sample output:
```Javascript
    [2017-12-05 14:19:32.396] [INFO] ACTIVE
    [2017-12-05 14:19:34.483] [INFO] UPDATED  14
    [2017-12-05 14:19:36.397] [INFO] COMPLETE:  {
    "ok": true,
        "start_time": "2017-12-05T18:43:50.948Z",
        "docs_read": 14,
        "docs_written": 14,
        "doc_write_failures": 0,
        "errors": [],
        "last_seq": <sequence-id>,
        "status": "complete",
        "end_time": "2017-12-05T19:19:36.397Z"
}
```

As seen above, the total duration is *35 minutes and 46 seconds*
There are no attachments in the docs and they have mere 25 columns. I believe the server-side `filter` approach is fast, but still the replication is way too long.

**Comment by daleharvey** [comment]
Where does the time go? it looks like it goes from active to complete in 4 seconds there, whats happening for the rest of the time?

**Comment by veetil09** [comment]
@daleharvey Sorry, I missed some code to point out the time difference:

```Javascript
const PouchDbMemory = require('pouchdb-memory');
const inmemeorydb = new PouchDbMemory(<remote-cloudant-db-name>);
logger.info(' remoteDbURL:  ', <remote-cloudant-db-url>);
inmemeorydb.replicate.from(
                <remote-cloudant-db-url>,
                {
                   filter: <filter-in-cloudant-design-doc>
            }).on('change', function (info) {
                logger.info(' UPDATED ', info.docs.length);
            }).on('active', function () {
                logger.info('ACTIVE');
            }).on('complete', function (info) {
                logger.info('COMPLETE: ', JSON.stringify(info, null, 2));
           });
```

Corresponding logging:
```
[2017-12-05 13:43:50.943] [INFO]  remoteDbURL:  <remote-cloudant-db-url>
[2017-12-05 14:19:32.396] [INFO] ACTIVE
[2017-12-05 14:19:34.483] [INFO] UPDATED  14
[2017-12-05 14:19:36.397] [INFO] PAUSED
[2017-12-05 14:19:36.397] [INFO]  COMPLETE:
```

The log stating the `remoteDbURL` is when I initiated the replication. As you can see, it takes around `35 minutes and 42 seconds` for the replication to become `Active` or even start. This is triggered during application startup and there is no interim processing. I believe when we start replication, it should start happening instantaneously but it doesn't (unless I am doing something wrong)

**Comment by daleharvey** [comment]
replication starts immediately, I would find out what its doing in those 35 minutes, is it writing deleted documents? are the requests taking an incredibly long time?

**Comment by veetil09** [comment]
@daleharvey I will have to insert breakpoint in the library and see what's going on. Perhaps I will turn on the `pouchdb` debugger and see what's happening and post it here.

**Comment by daleharvey** [comment]
Yeh the primary thing is to check the HTTP requests in the debugger, thats where most of the time will be spent, theres also https://pouchdb.com/api.html#debug_mode

**Comment by mohshraim** [comment]
@veetil09 what is the size of your db on cloudant? how many docs there?

**Comment by daleharvey** [comment]
Would need more info to debug here

**Comment by veetil09** [comment]
@mohshraim @daleharvey  Sorry for not responding for quite a while. I got it to work. But, I learnt that PouchDb does not support text-based indexes (like IBM Cloudant).

As a workaround, I switched to using the nodejs-cloudant that ships with the `follow` API that allows us to receive the `changes` (add/deletes/updates) made in the remote DB. Thanks for your help and directions. Really appreciate it.
